### PR TITLE
[6.3 Feature - Parameterized Subnets] UI Stubs

### PR DESCRIPTION
**Pull request by jyejare**
UI stubs for 6.3 feature 'Parameterized Subnets'

### CODE REVIEW

**Comment by jyejare** [comment]
@SatelliteQE/quality-engineers , I have raised stub tests for Parameterized Subnet feature. Please review the same.

**Comment by oshtaier** [comment]
sure, going to review today

File: tests/foreman/ui/test_subnet.py [hunk 1]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
...
+    @tier2
+    def test_positive_parameter_tab_presence(self):
+        """Presence of parameters tab in subnet
+
+        :id: beea7e8b-6d8e-40b0-bac1-0ad90b944088
+
+        :steps:
+
+            1. Go to Subnets component in infrastructure menu.
+            2. Attempt to create ne subnet.
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: `create one subnet`?

File: tests/foreman/ui/test_subnet.py [hunk 2]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
             self.subnet.update(name, new_subnet_mask=new_mask)
             result_object = self.subnet.search_and_validate(name)
             self.assertEqual(new_mask, result_object['mask'])
+
+
+class Parameterized_Subnet_Test_Cases(UITestCase):
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: `ParameterizedSubnetTestCase`

File: tests/foreman/ui/test_subnet.py [hunk 3]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
...
+            1. Go to Infrastructure -> Subnets.
+            2. Attempt to 'Create Subnet'
+            3. Update all the details for subnet creation
+            4. In Parameters tab, attempt to '+Add Parameter'.
+            5. Add parameter with single key and multiple values separated with
+                comma
+            6. Submit the changes for subnet with new parameter.
+
+        :expectedresults: The parameter with multiple values should be saved
+         in subnet
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]:
```
:expectedresults: The parameter with multiple values should be saved
    in subnet
```

File: tests/foreman/ui/test_subnet.py [hunk 4]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
...
+            2. Attempt to 'Create Subnet'
+            3. Update all the details for subnet creation
+            4. In Parameters tab, attempt to '+Add Parameter'
+            5. Add parameter with key having multiple names separated by valid
+                separators e.g fwd slash and value
+            6. Submit the changes for subnet with new parameter
+
+        :expectedresults: The parameter with multiple names separated by valid
+            separators should be saved in subnet
+
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: redundant line

File: tests/foreman/ui/test_subnet.py [hunk 5]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
...
+            2. Attempt to create ne subnet.
+            3. In create subnet dialog, check if Parameters tab is displayed.
+
+        :expectedresults: The parameter tab in subnet should be displayed
+            and available.
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_create_subnet_parameter(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: `test_positive_create_with_parameter`

File: tests/foreman/ui/test_subnet.py [hunk 6]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
...
+            4. In Parameters tab, attempt to '+Add Parameter'.
+            5. Add parameter with single key and single value.
+            6. Submit the changes for subnet with new parameter.
+
+        :expectedresults: The parameter should be saved in subnet
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_create_subnet_parameter_multiple_values(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: `test_positive_create_with_parameter_and_multiple_values`

File: tests/foreman/ui/test_subnet.py [hunk 7]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
...
+                comma
+            6. Submit the changes for subnet with new parameter.
+
+        :expectedresults: The parameter with multiple values should be saved
+         in subnet
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_create_parameter_multiple_names(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: `test_positive_create_with_parameter_and_multiple_names`

File: tests/foreman/ui/test_subnet.py [hunk 8]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
...
+                separators e.g fwd slash and value
+            6. Submit the changes for subnet with new parameter
+
+        :expectedresults: The parameter with multiple names separated by valid
+            separators should be saved in subnet
+
+        """
+    @stubbed
+    @tier1
+    def test_negative_create_parameter_multiple_names(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: `test_negative_create_with_parameter_and_invalid_separator`

File: tests/foreman/ui/test_subnet.py [hunk 9]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
...
+                invalid separators. e.g comma
+            6. Submit the changes for subnet with new parameter
+
+        :expectedresults: The parameter with multiple names separated by
+            invalid separators should not be saved in subnet
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_create_parameters(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: `test_positive_create_with_multuple_parameters`

File: tests/foreman/ui/test_subnet.py [hunk 10]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
...
+            6. Add more parameters with unique name and values
+            7. Submit the changes for subnet with all parameters
+
+        :expectedresults: The subnet should be created with multiple parameters
+            having unique names
+        """
+
+    @stubbed
+    @tier1
+    def test_negative_create_parameters(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: test_negative_create_with_duplicated_parameters

File: tests/foreman/ui/test_subnet.py [hunk 11]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
...
+            5. Go to that host parameters tab, under Global Parameters tab.
+
+        :expectedresults: The parameters from subnet should not be displayed
+            in host parameters tab
+        """
+
+    @stubbed
+    @tier2
+    def test_positive_subnet_parameters_override_from_host(self):
+        """Subnet parameters value can override from host
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: `Subnet parameters values can be overriden from host`

File: tests/foreman/ui/test_subnet.py [hunk 12]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
...
+                value
+            5. Submit the override the changes in subnet.
+
+        :expectedresults: The override value of subnet parameter from host
+            should not change actual value in subnet parameter
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_update_subnet_parameter(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: `test_positive_update_parameter`

File: tests/foreman/ui/test_subnet.py [hunk 13]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
...
+            5. Edit the above subnet and update parameter with new name and
+                value
+            6. Save subnet changes
+
+        :expectedresults: The parameter name and value should be updated
+        """
+
+    @stubbed
+    @tier1
+    def test_negative_update_subnet_parameter(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: `test_negative_update_parameter`

File: tests/foreman/ui/test_subnet.py [hunk 14]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
...
+                name. e.g name with comma or space
+            6. Save subnet changes
+
+        :expectedresults: The parameter should not be updated with invalid name
+        """
+
+    @stubbed
+    @tier2
+    def test_positive_update_subnet_parameter_host_impact(self):
+        """Subnet parameter update impact onto host
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: Clarify that statement
It is not obvious for me what `update impact onto host` means. How can we update impact?

File: tests/foreman/ui/test_subnet.py [hunk 15]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
...
+                subnet parameter
+            6. Save subnet changes
+
+        :expectedresults: The parameter should be deleted from subnet
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_delete_multiple_parameters(self):
+        """At a time, multiple subnet parameters can be deleted
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: `Multiple subnet parameters can be deleted at once`

File: tests/foreman/ui/test_subnet.py [hunk 16]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
...
+                more than one parameters
+            6. Save subnet changes
+
+        :expectedresults: Multiple parameters should be deleted from subnet
+        """
+
+    @stubbed
+    @tier2
+    def test_positive_delete_subnet_parameter_host_impact(self):
+        """Subnet parameter delete impact onto host
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: Clarify that statement

File: tests/foreman/ui/test_subnet.py [hunk 17]
```diff
@@ -293,3 +293,376 @@ def test_positive_update_mask(self):
...
+            7. Save subnet changes
+            8. Go to Host parameters tab
+
+        :expectedresults: The parameter should be deleted from host
+        """
+
+    @stubbed
+    @tier2
+    def test_positive_delete_subnet_parameter_overrided_host_impact(self):
+        """Subnet parameter override from host delete impact onto host
```
Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: Clarify that statement

**Comment by jyejare** [comment]
@oshtaier , Hey Thanks for your review. I resolved all your comments. Please take another look.

**Comment by jyejare** [comment]
@oshtaier sure, thanks!

**Pull request merged**